File: tests/fixtures/typescript_basic_shifted/src/widget.ts
// Widget module for the TypeScript scan fixture.
// This file has extra leading comments to test byte-shift stability.
// The logic below is identical to the non-shifted fixture.
import { EventEmitter } from "events";
import * as path from "path";
import type { Readable } from "stream";

export const LIMIT = 7;
export const NAME: string = "basic";

export interface Describable {
  describe(): string;
}

export type WidgetId = string;

export enum WidgetKind {
  Simple = "simple",
  Advanced = "advanced",
}

export class Base implements Describable {
  describe(): string {
    return "base";
  }
}

export class Widget extends Base {
  private id: WidgetId;
  kind: WidgetKind;

  constructor(value: number, kind: WidgetKind = WidgetKind.Simple) {
    super();
    this.id = String(value);
    this.kind = kind;
  }

  run(): number {
    return helper(Number(this.id));
  }

  describe(): string {
    return `widget-${this.id}`;
  }
}

export function helper(value: number): number {
  return value;
}

export function answer(): number {
  return new Widget(42).run();
}

export const arrowHelper = (x: number): number => x + 1;

File: tests/fixtures/python_basic/pkg/widget.py
"""Widget module for the Python scan fixture."""
from __future__ import annotations

import os
from collections import OrderedDict

LIMIT = 7
NAME: str = "basic"


class Base:
    def describe(self) -> str:
        return "base"


class Widget(Base):
    def __init__(self, value: int) -> None:
        self.value = value

    def run(self) -> int:
        return helper(self.value)


def helper(value: int) -> int:
    return value


def answer() -> int:
    return Widget(42).run()


def test_widget_runs():
    assert Widget(1).run() == 1


class TestWidget:
    def test_answer(self):
        assert answer() == LIMIT
